### PR TITLE
Retrieve product tags from product/tags REST api

### LINKED ISSUE #717: Tags from APIs attached to product
## Where did you encounter this bug?

* [x] Extractor
* [ ] Creator

## Describe the bug

Generated products.template.json contains resources product tags, which are fetched from the APIs.

Example 
```
Product (productname) has three api (api1, api2, api3)
api1 has tag1
api2 has tag2
api3 has tag1
```

Resulting products template will have the resources:
```
{
      "apiVersion": "2021-08-01",
      "type": "Microsoft.ApiManagement/service/products/tags",
      "name": "[concat(parameters('apimServiceName'), '/productname/tag1')]",
      "dependsOn": [
        "[resourceId('Microsoft.ApiManagement/service/products', parameters('apimServiceName'), 'productname')]"
      ],
      "properties": {
        "displayName": "test"
      }
}
```
It will be repeated for each tag from API per product, also apart from that the product template will be invalid because of having two identical resources product tags.

## What version of the DevOps Resource Kit is being used?

1.0.0-beta.4 and main branch

### PR DESCRIPTION
**Pull request by f-alizada**
Remove API tag dependency from product.

Previously generated product template contained the tags that were fetched from connected apis of the product, which lead to inconsistencies, and possible broken arm templates.

Closes #717

### CODE REVIEW

**Review by tomkerkhove — APPROVED** [review]

File: src/ArmTemplates/Common/API/Clients/Tags/TagClient.cs [hunk 1]
```diff
@@ -58,20 +58,12 @@ public async Task<List<TagTemplateResource>> GetAllTagsLinkedToApiAsync(string a
 
         public async Task<List<TagTemplateResource>> GetAllTagsLinkedToProductAsync(string productName, ExtractorParameters extractorParameters)
```
Comment (tomkerkhove) [inline src/ArmTemplates/Common/API/Clients/Tags/TagClient.cs:59]: Can we add some tests for this?

Comment (f-alizada) [reply src/ArmTemplates/Common/API/Clients/Tags/TagClient.cs:59]: In my opinion there is no such logic that should be covered by tests (unit), it should be done before where the collection of the tags was done. Right now, it is just http request to retrieve data.

Comment (DeagleGross) [reply src/ArmTemplates/Common/API/Clients/Tags/TagClient.cs:59]: Currently there are no E2E tests existing in resource-kit, Tom.
So I guess if we decide to add them at some moment - we will add the tests for all `<Entity>Client`s at once

File: src/ArmTemplates/Common/API/Clients/Tags/TagClient.cs [hunk 2]
```diff
@@ -58,20 +58,12 @@ public async Task<List<TagTemplateResource>> GetAllTagsLinkedToApiAsync(string a
...
         {
-            var apisLinkedToProduct = await this.apisClient.GetAllLinkedToProductAsync(productName, extractorParameters);
-
-            var tagTemplateResources = new List<TagTemplateResource>();
-            foreach (var productApi in apisLinkedToProduct)
-            {
-                var apiTags = await this.GetAllTagsLinkedToApiAsync(productApi.Name, extractorParameters);
+            (string azToken, string azSubId) = await this.Auth.GetAccessToken();
```
Comment (DeagleGross) [inline src/ArmTemplates/Common/API/Clients/Tags/TagClient.cs:61]: just fyi - you can simplify the expression
```suggestion
            var (azToken, azSubId) = await this.Auth.GetAccessToken();
```

**Review by DeagleGross — APPROVED** [review]

**Pull request merged**